extensions/authentication-jwt/src/services/jwt.service.ts

```diff
@@ -57,6 +59,7 @@ export class JWTService implements TokenService {
         'Error generating token : userProfile is null',
       );
     }
+    this.setCurrentUser(userProfile);
```

Comment:
This seems critical: set current user is done in the `authenticate` action, any reason we have to do it here again?

Reply:
Can u please put some light at code, I am not able to figure it out.